Compilation error in make check for  node_traversal_black.cpp

**Describe the bug**

When compiling cvc5 in debug mode and running `make check`, I see a compiler error related to
instantiating the `std::copy` template.

This happens with clang 15.0.0 on my Mac (running Mac OSX Ventura 13.6.1), but I don't think it's compiler
related. It's caused by the definition of the `std::copy` template in the LLVM libcxx:
(see https://github.com/llvm/llvm-project/blob/main/libcxx/include/__algorithm/copy.h)

Here's the compiler error.

```
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk/usr/include/c++/v1/__utility/pair.h:74:5: error: the parameter for this explicitly-defaulted copy constructor is const, but a member or base requires it to be non-const
    pair(pair const&) = default;
    ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk/usr/include/c++/v1/__algorithm/copy.h:111:1: note: in instantiation of template class 'std::pair<cvc5::internal::NodeDfsIterator, std::back_insert_iterator<std::vector<cvc5::internal::NodeTemplate<false>>>>' requested here
__copy(_InIter __first, _Sent __last, _OutIter __result) {
^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk/usr/include/c++/v1/__algorithm/copy.h:119:15: note: in instantiation of function template specialization 'std::__copy<std::_ClassicAlgPolicy, cvc5::internal::NodeDfsIterator, cvc5::internal::NodeDfsIterator, std::back_insert_iterator<std::vector<cvc5::internal::NodeTemplate<false>>>>' requested here
  return std::__copy<_ClassicAlgPolicy>(__first, __last, __result).second;
              ^
/Users/dutebrun/Repos/cvc5/test/unit/node/node_traversal_black.cpp:147:8: note: in instantiation of function template specialization 'std::copy<cvc5::internal::NodeDfsIterator, std::back_insert_iterator<std::vector<cvc5::internal::NodeTemplate<false>>>>' requested here
  std::copy(traversal.begin(), traversal.end(), std::back_inserter(actual));
       ^
1 error generated.

```

**Configure Options**

Here are the steps to reproduce this:
```
./configure.sh debug --auto-download --static
cd build
make
make check
```

The only fix I've found is to replace all the `std::copy` lines with explicit loops in `node_traversal_black.cpp`:
```
  std::vector<TNode> actual;
  for (auto node: traversal) {
    actual.push_back(node);
  }
```

--- a/src/expr/node_traversal.h
+++ b/src/expr/node_traversal.h
@@ -55,8 +55,8 @@ class NodeDfsIterator
   // Move/copy construction and assignment. Destructor.
   NodeDfsIterator(NodeDfsIterator&&) = default;
   NodeDfsIterator& operator=(NodeDfsIterator&&) = default;
-  NodeDfsIterator(NodeDfsIterator&) = default;
-  NodeDfsIterator& operator=(NodeDfsIterator&) = default;
+  NodeDfsIterator(const NodeDfsIterator&) = default;
+  NodeDfsIterator& operator=(const NodeDfsIterator&) = default;
   ~NodeDfsIterator() = default;
 
   // Preincrement
@@ -132,8 +132,8 @@ class NodeDfsIterable
   // Move/copy construction and assignment. Destructor.
   NodeDfsIterable(NodeDfsIterable&&) = default;
   NodeDfsIterable& operator=(NodeDfsIterable&&) = default;
-  NodeDfsIterable(NodeDfsIterable&) = default;
-  NodeDfsIterable& operator=(NodeDfsIterable&) = default;
+  NodeDfsIterable(const NodeDfsIterable&) = default;
+  NodeDfsIterable& operator=(const NodeDfsIterable&) = default;
   ~NodeDfsIterable() = default;
 
   NodeDfsIterator begin() const;

